src/fn_register.rs

```diff
@@ -156,10 +183,9 @@ pub fn by_value<T: Variant + Clone>(data: &mut Dynamic) -> T {
     mem::take(data).cast::<T>()
 }
 
-#[cfg(feature = "plugins")]
-impl<PL: Plugin> RegisterPlugin<PL> for Engine {
+impl<PL: crate::plugin::Plugin> RegisterPlugin<PL> for Engine {
```

Comment:
For consistent styling, let's import all the namespaces at the top of the file.  `import crate::plugin::Plugin;`